packages/grafana-ui/src/components/Combobox/Combobox.tsx

```diff
@@ -82,18 +115,27 @@ export const Combobox = ({ options, onChange, value, id, ...restProps }: Combobo
     isOpen,
     highlightedIndex,
     setInputValue,
-    selectItem,
     openMenu,
     closeMenu,
+    selectItem,
   } = useCombobox({
     inputId: id,
     items,
     itemToString,
     selectedItem,
+    onSelectedItemChange: ({ selectedItem, inputValue }) => {
+      onChange(selectedItem);
+    },
     defaultHighlightedIndex: selectedItemIndex ?? undefined,
     scrollIntoView: () => {},
     onInputValueChange: ({ inputValue }) => {
-      setItems(options.filter(itemFilter(inputValue)));
+      const filteredItems = options.filter(itemFilter(inputValue));
+      if (createCustomValue && inputValue && filteredItems.findIndex((opt) => opt.label === inputValue) === -1) {
+        setItems([...filteredItems, { label: inputValue, value: inputValue, description: 'Create custom' }]);
```

Comment:
Some i18n here would be good. I am open to suggestions for what the value should be. Also, at the moment this component really support descriptions, so we might want to remove description from the exposed `Option` and extend it with description locally.